go/cloud-query/internal/extractor/extractor_default.go

```diff
@@ -0,0 +1,80 @@
+package extractor
+
+import (
+	"fmt"
+
+	"github.com/lib/pq"
+	"github.com/samber/lo"
+	"k8s.io/klog/v2"
+
+	"github.com/pluralsh/console/go/cloud-query/internal/common"
+	"github.com/pluralsh/console/go/cloud-query/internal/connection"
+	"github.com/pluralsh/console/go/cloud-query/internal/log"
+	"github.com/pluralsh/console/go/cloud-query/internal/proto/cloudquery"
+)
+
+// DefaultExtractor is the default implementation of the Extractor interface.
+// It extracts data from a database connection and sends it to a sink.
+// It uses a map of resource factories to create resource objects from the extracted data.
+// It is designed to be used with a variety of cloud providers by providing the necessary resource factories.
+type DefaultExtractor struct {
+	sink      Sink
+	resources map[string]ResourceFactory
+}
+
+// Extract extracts data from the provided connection and sends it to the sink.
+func (in DefaultExtractor) Extract(conn connection.Connection) error {
+	klog.V(log.LogLevelDebug).InfoS("starting extraction")
+
+	for _, entry := range lo.Entries(in.Resources()) {
+		table := entry.Key
+		factory := entry.Value
+		columns, rows, err := conn.Query("SELECT * FROM" + pq.QuoteIdentifier(table))
+		if err != nil {
+			return fmt.Errorf("failed to query table '%s': %w", table, err)
+		}
+
+		klog.V(log.LogLevelDebug).InfoS("extracting table", "table", table, "columns", len(columns), "rows", len(rows))
+		for _, row := range rows {
+			klog.V(log.LogLevelTrace).InfoS("processing row", "table", table, "row", row)
+			rowJson, err := common.ToRowJSON(columns, row)
+			if err != nil {
+				return fmt.Errorf("failed to convert row to JSON: %w", err)
```

Comment:
I'm wondering if instead of breaking we should continue on these

Reply:
It would definitely be more resilient to random errors. In that case it makes sense to continue on error.